content/docs/atom-guide/dev-guide/shaders/azsl/unbounded-arrays.md

```diff
@@ -0,0 +1,306 @@
+---
+linktitle: Unbounded Arrays in AZSL
+title: AZSL, Binding Rules For Unbounded Arrays
+description: There are limitations into how many and what kind of Unbounded Arrays can be declared. 
+weight: 100
+---
+AZSL supports declaration of Unbounded Arrays inside `ShaderResourceGroup`s, but there are limitations into how many and what kind of Unbounded Arrays can be declared. The shader compiler, **AZSLc**, makes sure those rules are not violated.  
+  
+Here is a summary of the rules and limitations per platform (DX12, Vulkan and Metal). '--use-spaces' and '--unique-idx' are command line arguments for **AZSLc** that are used for each platform.  
+| --use-spaces   | --unique-idx | platform | Results |
+|----------------|--------------|----------|---------|
+| OFF | OFF |                   | Only the last SRG can contain one last unbounded array for each:<br> t[], u[], b[], s[]|
+| OFF | ON  |                   | Only the last SRG can contain one, and only one, last unbounded array across all resource types:<br>Only one of t[], u[], b[], s[]|
+| ON  | OFF | DX12              | Each SRG can contain one last unbounded array for each: <br> t[], u[], b[], s[]|
+| ON  | ON  | Vulkan<br> Metal  | Each SRG can contain one, and only one, last unbounded array across all resource types: <br> Only one of t[], u[], b[], s[]|
+  
+## Things to know before going over some examples:
+* "Register Space" in HLSL is the same as "Descriptor Set" in Vulkan.
+* When "--use-spaces" is used, **AZSLc** will assign a unique Register Space per SRG.
+* When "--unique-idx" is used, **AZSLc** will emit all resource descriptors in a Register Space to be in numerical sequence, regardless of the descriptor type. Platforms like Vulkan and Metal require this.
+* Unbounded arrays take over the whole range of register indices from the point they are being declared. This means two things:
+    1. It is possible to declare any kind of resources or any amount of bounded arrays of resources, BEFORE any given Unbounded Array of a particular kind.
+    2. Any resource declared AFTER an Unbounded Array will be considered an error.
+  
+## Examples
```

Comment:
### Note about my suggestions that follow
My suggestions in the following examples are to
* Restructure the hierarchy of headings, so that error case examples and solution examples are within the section of the example it's based on.
* Give each example a descriptive name. 

The overall heading structure that I propose looks like this: 

* Examples
  * Example: Register assignments
    * Error case
    * Error case
    * Solution
  * Example: Declaring unbounded arrays
  * Example: Resource assignments with `--unique-idx`
    * Error case
    * Solution
  * Example: Declaring unbounded arrays with `--unique-idx`

Reply:
Fixed.